### PR TITLE
Add a file upload for transcription

### PR DESCRIPTION
**Pull request by notwaldorf**
Disclaimer: I've only tried it with the sample wav we use but i think it works!

### CODE REVIEW

File: music/demos/transcription.ts [hunk 1]
```diff
@@ -59,36 +91,57 @@ async function transcribe(oaf: mm.OnsetsAndFrames, batchLength: number) {
...
+  const audio: AudioBuffer = await audioCtx.decodeAudioData(arrayBuffer);
+
+  const audioEl = document.getElementById('filePlayer') as HTMLAudioElement;
+  audioEl.hidden = false;
+  audioEl.src = window.URL.createObjectURL(blob);
+
   const oafA = new mm.OnsetsAndFrames(AUD_CKPT_URL);
-  const oafM = new mm.OnsetsAndFrames(MEL_CKPT_URL);
+  const start = performance.now();
   oafA.initialize()
```
Comment (adarob) [inline music/demos/transcription.ts]: Can we add something that signals to the users when the model is loading/loaded?

File: music/demos/transcription.html [hunk 2]
```diff
@@ -24,69 +33,85 @@ <h1>Onsets and Frames</h1>
...
+    <br>
+    <p><audio id="filePlayer" controls hidden></audio></p>
+    <p><b>Actual Transcription:</b> <code id="file-results" class="afileudio"></code></p>
+    <p><b>It Took:</b> <code id="file-time" class="file"></code></p>
+    <p><b>Total Leaked Memory:</b> <code id="file-leaked-memory"></code></b></p>
+  </section>
 
-  <h2>From Audio (250 frames / 8 seconds)</h2>
+  <h2>From a test local audio (250 frames / 8 seconds)</h2>
   <p>First we verify the model can transcribe a short sequence of piano audio, first computing its
```
Comment (adarob) [inline music/demos/transcription.html]: s/First//g

File: music/demos/transcription.ts [hunk 3]
```diff
@@ -59,36 +91,57 @@ async function transcribe(oaf: mm.OnsetsAndFrames, batchLength: number) {
...
+      await getAudioBufferFromBlob(blob) as ArrayBuffer;
+  const audio: AudioBuffer = await audioCtx.decodeAudioData(arrayBuffer);
+
+  const audioEl = document.getElementById('filePlayer') as HTMLAudioElement;
+  audioEl.hidden = false;
+  audioEl.src = window.URL.createObjectURL(blob);
+
   const oafA = new mm.OnsetsAndFrames(AUD_CKPT_URL);
-  const oafM = new mm.OnsetsAndFrames(MEL_CKPT_URL);
+  const start = performance.now();
```
Comment (adarob) [inline music/demos/transcription.ts]: start this after load

File: music/demos/transcription.ts (hunk 1 above)
Comment (notwaldorf) [reply music/demos/transcription.ts]: we could, but we don't do this for any of the other demos rn.

File: music/demos/transcription.html [hunk 4]
```diff
@@ -24,69 +33,85 @@ <h1>Onsets and Frames</h1>
...
+    <p><audio id="filePlayer" controls hidden></audio></p>
+    <p><b>Actual Transcription:</b> <code id="file-results" class="file"></code></p>
+    <p><b>It Took:</b> <code id="file-time" class="file"></code></p>
+    <p><b>Total Leaked Memory:</b> <code id="file-leaked-memory"></code></b></p>
+  </section>
 
-  <h2>From Audio (250 frames / 8 seconds)</h2>
-  <p>First we verify the model can transcribe a short sequence of piano audio, first computing its
-    <a href="https://librosa.github.io/librosa/generated/librosa.feature.melspectrogram.html" target="_blank"></a>>mel spectogram</a>.
+  <h2>From a test local audio (250 frames / 8 seconds)</h2>
```
Comment (adarob) [inline music/demos/transcription.html]: From test audio

File: music/demos/transcription.ts (hunk 1 above)
Comment (adarob) [reply music/demos/transcription.ts]: Please do since this one takes a while.

**Pull request merged**